curated_transformers/models/pytorch/embeddings.py

```diff
@@ -28,3 +29,57 @@ def forward(self, x: Tensor) -> Tensor:
             x - (batch, seq_len)
         """
         return self.pe[x.size(1), :]
+
+
+class RotaryEmbeddings(Module):
+    cos: Tensor
+    sin: Tensor
+    theta: Tensor
+
+    def __init__(self, width: int, seq_len: int = 512, *, base: int = 10000):
+        """Cached rotary embeddings (Su et al., 2021). The rotary embedding
+        will be computed for up to 'initial_len' positions. However, the
+        embedding will be recomputed when a longer sequence is found in the
+        input.
+
+        width (int):
+            Rotary embedding dimensionality.
+        seq_len (int):
+            Number of positons to initially pre-cache.
+        base (int):
+            The base used for Θ_i, determines the cycle length of the
+            embeddings."""
+        super().__init__()
+
+        # Θ_i = 10000^(-2(i-1)/d)
+        theta = torch.pow(base, -torch.arange(0, width, 2, dtype=torch.float) / width)
```

Comment:
We should add a check to ensure that `width` is even (or else we break at the `view()` call in line 68).